[http2] Http2TestTlsTwoRouter.test_head_request_no_alpn_no_http2_prior_knowledge failing

```
self = <Popen: returncode: 28 args: ['curl', '-q', 'https://localhost:24993/images/...>, endtime = 833.901449485, orig_timeout = 300.0, stdout_seq = []
stderr_seq = [b'  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 D...pper-router/skupper-router/skupper-router/tests/ssl_certs/ca-certificate.pem\n  CApath: none\n', b'} [5 bytes da', ...]
skip_check_and_raise = False

    def _check_timeout(self, endtime, orig_timeout, stdout_seq, stderr_seq,
                       skip_check_and_raise=False):
        """Convenience for checking if a timeout has expired."""
        if endtime is None:
            return
        if skip_check_and_raise or _time() > endtime:
>           raise TimeoutExpired(
                    self.args, orig_timeout,
                    output=b''.join(stdout_seq) if stdout_seq else None,
                    stderr=b''.join(stderr_seq) if stderr_seq else None)
E           subprocess.TimeoutExpired: Command '['curl', '-q', 'https://localhost:24993/images/test.jpg', '--verbose', '--cacert', '/home/runner/work/skupper-router/skupper-router/skupper-router/tests/ssl_certs/ca-certificate.pem', '--cert-type', 'PEM', '--cert', '/home/runner/work/skupper-router/skupper-router/skupper-router/tests/ssl_certs/client-certificate.pem:client-password', '--key', '/home/runner/work/skupper-router/skupper-router/skupper-router/tests/ssl_certs/client-private-key.pem', '--output', '/__w/skupper-router/skupper-router/skupper-router/build/tests/system_test.dir/tests/system_tests_tcp_adaptor_tls/HttpOverTcpTestTlsTwoRouterNginx/setUpClass/test.jpg', '--http1.1', '--max-time', '300.0']' timed out after 300.0 seconds
```

https://github.com/skupperproject/skupper-router/actions/runs/4078785932/jobs/7029415128#step:30:4437

--- a/tests/system_tests_http2_tls.py
+++ b/tests/system_tests_http2_tls.py
@@ -220,14 +220,22 @@ class Http2TestTlsTwoRouter(Http2TestTwoRouter, RouterTestSslBase):
         # This means that curl client will not offer an ALPN protocol at all. The router (server) sends back 'h2'
         # in its ALPN response which is ignored by the curl client.
         # The TLS handshake is successful and curl receives a http2 settings frame from the router which it does
-        # not understand. Hence it complains with the error message
+        # not understand. Hence it complains with either of the following error messages based on the version of
+        # openssl.
         # 'Received HTTP/0.9 when not allowed'
+        # 'Weird server reply'
+        # The main goal of this test is to make sure the router closes the connection
+        # when curl speaks HTTP/1.1 because the http2 adaptor does not speak HTTP/1.1
         address = self.router_qdra.http_addresses[0]
         _, out, err = self.run_curl(address,
                                     args=self.get_all_curl_args(['--head']),
                                     http2_prior_knowledge=False,
                                     no_alpn=True, assert_status=False)
-        self.assertIn('Received HTTP/0.9 when not allowed', err)
+        self.assertEqual(out, '', f"Expected value of out to be an empty string but got {out}")
+        test_passed = False
+        if 'Received HTTP/0.9 when not allowed' in err or 'Weird server reply' in err:
+            test_passed = True
+        self.assertTrue(test_passed, f"unexpected curl error: {err}")
 
 
 class Http2TlsQ2TwoRouterTest(RouterTestPlainSaslCommon, Http2TestBase, RouterTestSslBase):

